crates/stc_ts_file_analyzer/src/analyzer/expr/mod.rs

```diff
@@ -4355,6 +4355,47 @@ impl Analyzer<'_, '_> {
         if should_be_optional && include_optional_chaining_undefined {
             Ok(ty.union_with_undefined(span))
         } else {
+            if self.rule().no_unchecked_indexed_access && matches!(self.ctx.pat_mode, PatMode::Decl) {
+                let field_is_symbol = match &prop {
+                    Key::Computed(key) => key.ty.is_symbol_like(),
+                    _ => false,
+                };
+
+                let is_variadic_tuple = match &obj_ty {
+                    Type::Arc(_) => match obj_ty.normalize() {
```

Comment:
I think this can be problematic in the future, as `Type::Arc` should be used in more cases and it's not for _variadic_. That's why I always use `Type::normalize` while pattern matching